code/modules/projectiles/guns/energy/lasmusket.dm

```diff
@@ -0,0 +1,326 @@
+	if(!lens_secure && loadedassembly)
+		to_chat(user, "<span class='warning'>\The [loadedassembly] inside \the [src] is unsecured.</span>")
+
+/obj/item/weapon/gun/energy/lasmusket/afterattack(atom/A, mob/living/user, flag, params, struggle = 0)
+	if (istype(A, /obj/item/weapon/storage/backpack ))
+		return
+	else if (A.loc == user.loc)
+		return
+	else if (A.loc == user)
+		return
+	else if (locate (/obj/structure/table, src.loc))
+		return
+	if(!cell_secure || !lens_secure)
+		click_empty(user)
+		return
+	else if(cell_secure)
+		if(loadedcell.charge <=0)
+			click_empty(user)
+			return
+	if(flag)
+		return //we're placing gun on a table or in backpack
+	if(harm_labeled >= min_harm_label)
+		to_chat(user, "<span class='warning'>A label sticks the trigger to the trigger guard!</span>")//Such a new feature, the player might not know what's wrong if it doesn't tell them.
+		return
+	if(round(loadedcell.charge < 5000))
+		to_chat(user, "<span class = 'warning'>\The [src] doesn't have enough energy to fire!</span>")
+		return
+	calculate_strength(A,user,params, "struggle" = struggle)
+
+/obj/item/weapon/gun/energy/lasmusket/proc/calculate_strength(atom/A as mob|obj|turf|area, mob/living/user as mob|obj, flag, params, struggle = 0)
+	if(!loadedcell || !loadedassembly || loadedcell.charge <= 0) //Probably shouldn't happen
+		return
+	var/strength = round(loadedcell.charge)
+	if(strength)
+		switch(strength)
+			if(5000 to 9999)
+				projectile_type = /obj/item/projectile/beam/weaklaser
+			if(10000 to 19999)
+				projectile_type = /obj/item/projectile/beam/lightlaser
+			if(20000 to 29999)
+				projectile_type = /obj/item/projectile/beam/
+			if(30000 to 49999)
+				projectile_type = /obj/item/projectile/beam/captain
+			if(50000 to INFINITY)
+				projectile_type = /obj/item/projectile/beam/heavylaser
+
+		if(Fire(A,user,params, "struggle" = struggle))
+			if(strength > 50000 && !flawless)
+				to_chat(user, "<span class='warning'>\The [loadedassembly] inside \the [src] melts!</span>")
+				to_chat(user, "<span class='warning'>\The [loadedcell] inside \the [src]'s power bank ruptures!</span>")
+				qdel(loadedassembly)
+				loadedassembly = null
+				lens_secure = 0
+				qdel(loadedcell)
+				loadedcell = null
+				cell_secure = 0
+			else if (!flawless)
+				loadedassembly.durability -= (strength/2000) //Lens assembly degrades with each shot. Ultra cell gives 4 shots.
+				if(loadedassembly.durability <= 0)
+					to_chat(user, "<span class='warning'>\The [loadedassembly] inside \the [src] [strength > 100 ? "shatters under" : "finally fractures from"] the stress!</span>")
+					qdel(loadedassembly)
+					loadedassembly = null
+					lens_secure = 0
+			fire_sound = initial(fire_sound)
+			loadedcell.charge = 0 //Drain the cell after firing
+		update_icon()
+
+/obj/item/weapon/gun/energy/lasmusket/attack(mob/living/M, mob/living/user, def_zone) //Suicide handling to make sure the gun's empty afterward.
```

Comment:
make sure this is thoroughly tested (e.g.: start suicide but cancel, try pressing a button with the gun)